### ISSUE TITLE
Turbopack support

### ISSUE THREAD

**Issue by arlyon** [fuma-nama/fumadocs#459]
### What problem will this feature address?

Turbopack in nextjs is enjoyable in dev and will soon be supported for builds. It would be nice for fumadocs to support this, and it does.. mostly. The missing part is fumadocs-mdx

### Describe the solution you'd like

We need a few changes to support this:

- turbo plugins are currently configured in a different section (`experimental.turbo`) in the next conf
- turbo supports all the parts needed, we just need to plumb them slightly differently
- passing functions / closures across the the turbo boundary is unsupported since it needs to serialize to json, invoke turbo, which runs the plugins. this is problematic for our set up, so I propose a syntax of ["plugin-name", { settings}] to allow turbo to import and evaluate the plugin, instead. This restriction will likely be a breaking change for people wanting to migrate to turbo dev.

Since build is not turbo-based, we will still need to include the webpack config.

**TLDR** createMDX will need to create the webpack plugins on turbo's behalf on the 'other side' since passing closures through turbopack doesn't work.

### Describe alternatives you've considered

I have a patch/hack that is working locally. I would like to upstream it. Just thought I'd open an issue here first.

**Comment by fuma-nama** [comment]
Welcome to provide a PR! I'm going to work on that once they've released the Plugin API, it would be great if we can support Turbopack sooner.

**Comment by fuma-nama** [comment]
Fumadocs MDX v10 and Content Collections will work on Turbopack.

For Fumadocs MDX: Some performance optimizations might not work because of https://github.com/vercel/next.js/issues/69502, Webpack is still recommended.

I'm looking for solutions to load TypeScript configs with Turbopack (currently using esbuild), would be nice to remove more dependencies.
